src/helpers/GiveRedemptionBalanceHelper.ts

```diff
@@ -18,33 +17,58 @@ interface IDonorAddresses {
 // Gets a recipient's info. Separating it out into a helper allows us to call it on addresses
 // other than the current user's without overwriting the data in Redux store. This is needed
 // to pull data on our partner projects
+/**
+ * Gets a recipient's donation info including redeemable value (denominated in sOHM),
+ * current sOHM principal directed towards them earning yield, and the gOHM equivalent
+ * value of the principal directed towards them
+ * @param address The address of the user or recipient we wish to pull this info for
+ * @param networkID ID number of network the user is currently connected to
+ * @param provider Ethereum network provider object
+ * @returns Redeemable sOHM plus recipientInfo object (totalDebt and agnosticDebt)
+ */
 export const getRedemptionBalancesAsync = async ({ address, networkID, provider }: IBaseAddressAsyncThunk) => {
   let redeemableBalance = 0;
   const recipientInfo: IUserRecipientInfo = {
-    totalDebt: "",
-    carry: "",
-    agnosticDebt: "",
-    indexAtLastChange: "",
+    totalDebt: "0",
+    agnosticDebt: "0",
   };
 
-  if (addresses[networkID] && addresses[networkID].GIVING_ADDRESS) {
+  if (!(addresses[networkID] && addresses[networkID].GIVING_ADDRESS)) {
+    console.log("Unable to find MOCK_SOHM contract on chain ID " + networkID);
+  } else {
+    const gohmContract = new ethers.Contract(addresses[networkID].GOHM_ADDRESS as string, gOHM, provider);
     const givingContract = new ethers.Contract(addresses[networkID].GIVING_ADDRESS as string, OlympusGiving, provider);
-    redeemableBalance = await givingContract.redeemableBalance(address);
+
+    // Get current redeemable balance across all deposits to the user. This is
+    // returned in gOHM, so it has to be converted from gOHM to sOHM to be
+    // represented in the frontend
+    const gohmRedeemable = await givingContract.totalRedeemableBalance(address);
+    redeemableBalance = await gohmContract.balanceFrom(gohmRedeemable);
 
     try {
-      const recipientInfoData = await givingContract.recipientInfo(address);
-      recipientInfo.totalDebt = ethers.utils.formatUnits(recipientInfoData.totalDebt.toNumber(), "gwei");
-      recipientInfo.carry = ethers.utils.formatUnits(recipientInfoData.carry.toNumber(), "gwei");
-      recipientInfo.agnosticDebt = ethers.utils.formatUnits(recipientInfoData.agnosticDebt.toNumber(), "gwei");
-      recipientInfo.indexAtLastChange = ethers.utils.formatUnits(
-        recipientInfoData.indexAtLastChange.toNumber(),
-        "gwei",
-      );
+      const recipientIds = await givingContract.getRecipientIds(address);
```

Comment:
Why is this in try/catch but the `givingContract` and `gohmContract` calls aren't?

Reply:
because we've already determined that givingContract exists, so there's no concern about instantiating it, and totalRedeemableBalance can't revert. getRecipientIds can return a null value though which then could create problems in future calls. I don't mind adding everything in the try/catch if you think it makes the code neater to look at/consistent